### ISSUE TITLE
v2.1.4 breaks backwards compatibility with v2.1.3

### ISSUE THREAD

**Issue by dmcnally** [rubysherpas/paranoia#276]
The following addition https://github.com/radar/paranoia/compare/v2.1.3...v2.1.4#diff-ccbdb603cb37c0f5ccb9d8b62932dee4R258 breaks backwards compatibility with v2.1.3

Previous behaviour validated uniqueness constraints across the entire dataset of deleted and non deleted items.

New behaviour seems to exclude deleted items from uniqueness validations.

There doesn't seem to be any way to selective revert to previous behaviour.

**Comment by radar** [comment]
Ouch. I'd consider this serious enough to revert it in a commit, and release 2.1.5 with that revert, then to have a discussion on the original issue about whether or not that should be within paranoia. I believe the answer would be "yes it should be", but we'll probably need it to be opt-in.

Please submit a PR which reverts that behaviour. I will review it and then if it's acceptable, merge it then release 2.1.5.

**Comment by dmcnally** [comment]
Please see PR https://github.com/radar/paranoia/pull/277

**Comment by laffy** [comment]
It would nice nice to include a flag when defining a model as `acts_as_paranoid` so that you could choose whether to validate archived records or not.

**Comment by dmcnally** [comment]
@laffy I don't think that will work.  You need to specify at the validation level if the deleted items should be included or not.

Our situation has us requiring 
- a unique reference that covers all objects
- a second unique constraint covers non-deleted objects

This allows us to have a unique reference that can be used to find archived items but reuse the same titles/names etc for non-archived.

**Comment by donv** [comment]
This is biting us now:  We need uniqueness validation of a string column across all records to be able to restore a record and still be sure the column is unique.

This works fine with paranoia 2.1.3 but fails with 2.1.4 and 2.1.5.

We can accept that the default is to only validate uniqueness across non-deleted records, but we need a way to specify per validation rule wether uniqueness should be across all records or across non-deleted records only.
